generator/src/main/java/com/scottlogic/deg/generator/inputs/AtomicConstraintReaderLookup.java

```diff
@@ -299,18 +302,26 @@ private static Object potentialUnwrapDate(Object value) throws InvalidProfileExc
         return parseDate((String)date);
     }
 
-    private static LocalDateTime unwrapDate(Object value) throws InvalidProfileException {
+    private static OffsetDateTime unwrapDate(Object value) throws InvalidProfileException {
         Object date = potentialUnwrapDate(value);
-        if (date instanceof LocalDateTime)
-            return (LocalDateTime) date;
+        if (date instanceof OffsetDateTime)
+            return (OffsetDateTime) date;
 
         throw new InvalidProfileException(String.format("Dates should be expressed in object format e.g. { \"date\": \"%s\" }", value));
     }
 
-    static LocalDateTime parseDate(String value) throws InvalidProfileException {
-        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("u-MM-dd'T'HH:mm:ss'.'SSS").withResolverStyle(ResolverStyle.STRICT);
+    static OffsetDateTime parseDate(String value) throws InvalidProfileException {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("u-MM-dd'T'HH:mm:ss'.'SSS")
+            .withResolverStyle(ResolverStyle.STRICT)
+            .withZone(ZoneOffset.UTC);
+
         try {
-            LocalDateTime parsedDateTime = LocalDateTime.parse(value, formatter);
+            OffsetDateTime parsedDateTime = ZonedDateTime.parse(
+                value.endsWith("Z")
```

Comment:
All attempts to make the `Z` suffix optional via the pattern and the `DateTimeFormatterBuilder` failed, this is the only way i could seem to get it to work, I'm all ears if someone has a better way of getting it to work.

Reply:
Should we be using OffsetDateTime.parse here instead.

would that work with the z?

Reply:
OffsetDateTime requires the offset to be in the form `+HH:mm` which we're not supporting, at this stage.

Reply:
Just tried again and no-dice, the `X` formatter item cannot be used when milliseconds are included, according to this answer.

https://stackoverflow.com/a/29210140

Including the following formats in the pattern do not work:
- `Z` or `[Z]`
- `X`, `[X]`, `XXX` or `[XXX]`

Reply:
I pushed a commit that handles that Z parsing in a more Java-idiomatic way, check it out :+1: